templates/include/yarp/ast.h.erb

```diff
@@ -46,26 +46,31 @@ typedef struct yp_node_list {
     size_t capacity;
 } yp_node_list_t;
 
-typedef enum {
+enum yp_node_type {
 <%- nodes.each_with_index do |node, index| -%>
     <%= node.type %> = <%= index + 1 %>,
 <%- end -%>
-} yp_node_type_t;
+};
+
+typedef uint16_t yp_node_type_t;
+typedef uint16_t yp_node_flags_t;
 
 // We store the node type enum in every node in the tree. We don't have nearly
 // as many node types as we do bits in an enum, so we're going to use the top
 // half of the enum to store flags about the node.
-#define YP_NODE_FLAGS_SHIFT (sizeof(yp_node_type_t) * 8 / 2)
-#define YP_NODE_FLAG_NEWLINE (1U << YP_NODE_FLAGS_SHIFT)
+static const unsigned int YP_NODE_FLAG_NEWLINE = 0x1;
 
 // For easy access, we define some macros that manipulate only the bottom half
 // of the node type enum.
```

Comment:
It would be best to update these comments to reflect the new way to represent flags